### ISSUE TITLE
Irreverent - Ecowitt is now integrated officially to Home-assistant

### ISSUE THREAD

**Issue by dopeytree** [garbled1/homeassistant_ecowitt#176]
Heads up if you come here wondering why this HACs integration isn't working. 
Delete it and reboot HA.

Ecowitt is now integrated officially to Homeassistant.
So after rebboo go to integrations --> add --> ecowit
You'll then go to your ecowit device and add in some code

**Comment by spacelama** [comment]
Of course, if you already had devices and entities configured, it's not really as simple as this, is it?

Is there a simple way to remove the integration but keep all the enumeration of your devices the same?

**Comment by dopeytree** [comment]
Not sure but lets explore.
What do your entities look like? sensor name etc

I have a GW1100A 

Sensors show up like this:
sensor.gw1100a_absolute_pressure

**Comment by Megachip** [comment]
Is there  a migration path?

**Comment by Docsharp00** [comment]
Hi there, was someone able to migrate the devices to HA integration? And 8s there a way to not loose the history?

**Comment by mr-p666** [comment]
Yes, it worked without any problems.
Since I've been using https directly up until now, I had to activate nginx, but that wasn't a problem at all.
All the historical data was retained. 👍🏼

**Comment by guidocioni** [comment]
> Yes, it worked without any problems. Since I've been using https directly up until now, I had to activate nginx, but that wasn't a problem at all. All the historical data was retained. 👍🏼

How did you retain the history? I've been maintaining this integration locally by fixing all the deprecations manually because I really don't want to go to the process of having to add and rename every entity manually. I don't think there's an automatic way to do the migration

**Comment by mr-p666** [comment]
Luckily I didn't have to do anything else. It seems that the integrations use the same device IDs, which meant there was nothing to do and I could use my old dashboards, as well as the history from the old integration.

**Comment by spacelama** [comment]
Keep and test backups to roll back to. And test straight away, not days later when you're doing a bunch of other upgrades and have made changes you don't want to lose when you need to roll back (have I ever said I hate autodiscovered systems that don't keep an entire infrastructure-as-code representation of themselves?). Mine *didn't* work, and I don't know what I did wrong and how I recovered it.

**Comment by guidocioni** [comment]
Yeah, on my side, as long as it is allowed, I'm going to continue using the non-official ecowitt integration by fixing manually deprecations/errors as long as I can. It is just not worth to migrate everything to the official one and suddenly lose the history or having to perform weird workarounds.

**Comment by Megachip** [comment]
> Is there a migration path?

Cause the responses here are a bit confusing... Is there a save way of migration without loosing more than 2 years of data? Will I loose any features?
